src/libsyntax/errors/emitter.rs

```diff
@@ -23,6 +22,23 @@ use std::io;
 use std::rc::Rc;
 use term;
 
+#[derive(Debug)]
+enum Error {
+    Io(io::Error),
+    Term(term::Error),
+}
+
+impl From<io::Error> for Error {
+    fn from(err: io::Error) -> Self {
+        Error::Io(err)
+    }
+}
+
+impl From<term::Error> for Error {
+    fn from(err: term::Error) -> Self {
+        Error::Term(err)
+    }
+}
```

Comment:
Personally, I'd prefer the following so we don't loose any information:

```rust
impl From<term::Error> for io::Error {
    fn from(err: term::Error) -> io::Error {
        let kind = match &err {
            &term::Error::Io(ref e) => e.kind(),
            _ => ErrorKind::Other,
        };
        io::Error::new(kind, err)
    }
}
```

This way, we keep the fact that this IO error was caused by a term error.